### DISCUSSION TITLE
Making a Query on an Access adn Oracle Join

### DISCUSSION THREAD

**Discussion by mmitsch** [dbeaver/dbeaver#18633] [General]
I am REALLY new to dbeaver (just bought it) and am slowly learing the interface -- like it so far!

I have a couple of questions that  could use some help with.

I had some data in Excel that I wanted to compare against the Oracle main app on our server.  I figured that if  could get the data to Access that maybe I could see it in dbeaver and then make a sql guery to look of matches.  

I can see the access data in dbeaver, as well as the oracle data.  I was even albe to make an ERD diagram on how I would join them.  When I create a new SQL query, however, I cannot seem to do an inner or outer join from a table in one connection source (Access) to the other (Oracle).  Is there something I'm mssing?

And, is the ER diagram at all useful in defingh the mappng to help the query or is it just for keeping record of the didagram?

Let me know and thanks!

**Comment by SanjayPethe** [comment]
I have the same question:
Can you do cross database joins in the DBeaver Enterprise edition? So for example, can I write a query joining a table in my MS Access database to one in my SQL Server or Teradata database? If yes, can you point me to the documentation on how to do that?

To my knowledge this cannot be done in any edition of DBeaver. You can however do it in MS Access through an ODBC connection to the Oracle database. I have used MS Access many times for such cross database comparisons (but it can be slow)!

**Reply by LonwoLonwo** [reply]
> So for example, can I write a query joining a table in my MS Access database to one in my SQL Server or Teradata database?

Unfortunately, DBeaver doesn't have such an option.

**Reply by SanjayPethe** [reply]
Thanks. DBeaver support suggested an alternative using Trino. You would setup Trino to connect to the different databases (MS Access, Oracle, Teradata, Snowflake, etc) and then a single DBeaver connection to Trino, which would then allow it so see all those databases. You can then join tables across databases. I have not tried this yet.

**Reply by LonwoLonwo** [reply]
Hmm, interesting idea. Probably Presto also can be used in this case.
